### ISSUE TITLE
Remove default prefix

### ISSUE THREAD

**Issue by Jamie-SA** [semanticarts/gist#898]
I seem to remember seeing something that recommended that ontology files should not have default prefix statements (the blank or `:` namespace). I don't know why we'd want to use one in gist but I see one in gistCore. I suggest we remove it.

**Comment by rjyounes** [comment]
Agreed. @dylan-sa and I observed this somewhere and wondered if Protege was adding it. If so, we could include removing it in the pre-commit hook that deletes the skos annotation declarations.

**Comment by rjyounes** [comment]
We would need a process that runs after the serializer to remove the base declaration. Could include in the serializer pre-commit. 

DECISION: Include in the serializer pre-commit.

**Comment by rjyounes** [comment]
@sa-bpelakh Please confirm that this is indeed inserted by the serializer before you begin this implementation. Possibly I'm misremembering.

**Comment by rjyounes** [comment]
@uscholdm Do we want to ask the EDM Council to add this as an option to the serializer?

**Comment by uscholdm** [comment]
It worked before, so it might work again - especially since use of Protege is so widespread. 

We'd need to have a clear example to hand them. A ton of experiments were done, that would have to be condensed, clear and concise.

**Comment by sa-bpelakh** [comment]
> @sa-bpelakh Please confirm that this is indeed inserted by the serializer before you begin this implementation. Possibly I'm misremembering.

I checked, and the serializer does not re-add the default namespace if it is removed manually. If I modify the commit hook to remove it anyhow, and then push in a single commit removing it from every ontology, we won't even need to have it as a step in the build - unless you think someone out there is bypassing the commit hook.

**Comment by uscholdm** [comment]
> unless you think someone out there is bypassing the commit hook.

That could happen if a setup is not done correctly or gets corrupted.

**Comment by rjyounes** [comment]
Create new issue to have the pre-commit hook remove it. For this release, Dylan will remove it before release.

**Comment by Jamie-SA** [comment]
I think the requirement is for the pre-commit hook to remove any default prefix in any ontology file.

**Comment by stevenchalem** [comment]
My vote is to use Python. In case it's useful, I wrote the following. I think it does the job and I have tested it on Windows (as a standalone Python script, not as part of a pre-commit hook.)

```python
#!/usr/bin/env python3
import re, sys, pathlib

# Regex matches both Turtle (@prefix) and SPARQL (PREFIX) default declarations.
pattern = re.compile(
    r'(?mi)^\s*(?:@prefix|prefix)\s*:\s*<[^>]+>\s*(?:\.\s*)?(?:#.*)?$'
)

changed = False
for filePath in pathlib.Path("ontologies").glob("*.ttl"):
    currentText = filePath.read_text(encoding="utf-8")
    newText = pattern.sub("", currentText).strip() + "\n"
    if newText != currentText:
        filePath.write_text(newText, encoding="utf-8")
        changed = True

# Exit 1 if anything changed, so you re-stage before committing.
# If no change was needed, exit 0 and let the commit proceed.
sys.exit(1 if changed else 0)
```

**Comment by rjyounes** [comment]
@Jamie-SA Thanks, I misread Boris's suggestion.
